### PR TITLE
feat: implement new Fuel Wallet Connector

### PR DESCRIPTION
**Pull request by luizstacio**
This PR includes the new changes on the Fuel Wallet SDK, to support the new Connectors API dropping support for the old window.fuel injection.

- https://github.com/FuelLabs/fuels-wallet/pull/966
- https://github.com/FuelLabs/fuels-wallet/pull/972
- https://github.com/FuelLabs/fuels-wallet/pull/982
- https://github.com/FuelLabs/fuels-wallet/pull/1006
- https://github.com/FuelLabs/fuels-wallet/pull/984

⚠️ CI: Check links will be fixed once the PR is merged, due to the requirement to have all links working, but the current links point to files on master that do not exist.

### CODE REVIEW

File: packages/docs/src/hooks/useAssets.tsx [hunk 1]
```diff
@@ -19,11 +19,6 @@ export function useAssets() {
 
     // Query assets
     queryAssets();
-
-    fuel.on('assets', queryAssets);
```
Comment (matt-user) [inline packages/docs/src/hooks/useAssets.tsx:23]: why did we remove this?

**Review by matt-user — CHANGES_REQUESTED** [review]
Viewing the asset page on the docs does not work.  Here is the error message.
Screenshot 2024-01-19 at 9 39 05 AM

I tested everything except for the assets page in the docs, and it all worked.

File: packages/app/src/systems/Transaction/components/TxHeader/TxHeader.test.tsx [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import { render, testA11y, screen, act, fireEvent } from '@fuel-ui/test-utils';
-import { BLOCK_EXPLORER_URL } from '@fuel-wallet/sdk';
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/app/src/systems/Transaction/components/TxHeader/TxHeader.test.tsx:2]: It's not something that should be inside the wallet sdk this is why was removed

File: packages/docs/src/hooks/useAssets.tsx (hunk 1 above)
Comment (luizstacio) [reply packages/docs/src/hooks/useAssets.tsx:23]: This file is not used anymore I had delete it

**Comment by luizstacio** [comment]
> Viewing the asset page on the docs does not work. Here is the error message. Screenshot 2024-01-19 at 9 39 05 AM
> 
> I tested everything except for the assets page in the docs, and it all worked.

I had to revert your changes on the assets lib, the issue is due to the implementation of the assets package not returning the networks inside the processed asset. This could be fixed on the next version.

**Comment by LuizAsFight** [comment]
getting this error when trying to transfer


image

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Comment by LuizAsFight** [comment]
getting this error when testing with live link

image

File: packages/app/playwright/crx/crx.test.ts [hunk 3]
```diff
@@ -30,6 +30,9 @@ import {
 
 const WALLET_PASSWORD = 'Qwe123456$';
 
+/**
+ * @todo: skip e2e tests of the wallet application for implement it on the next PR
+ */
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/app/playwright/crx/crx.test.ts:35]: I forgot to remove the comment, I already remove the skip

File: packages/app/src/systems/Core/utils/assets.tsx [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
+export const fuelAssets = assets.map((asset) => {
+  const fuelNetworkAsset = asset.networks.find(
+    (n) => n.type === 'fuel'
+  ) as Fuel;
+  return {
+    ...asset,
+    assetId: fuelNetworkAsset.assetId,
+    decimals: fuelNetworkAsset.decimals,
+  };
+});
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/app/src/systems/Core/utils/assets.tsx:13]: We can't here, I just revert the changes. We need a change on the util to make it work, it should return the networks array also, and currently it dosen't.

[the remark this answers is not in the document]
Comment (LuizAsFight) [reply packages/app/src/systems/Core/utils/assets.tsx:13]: sg

**Review by LuizAsFight — DISMISSED** [review]

**Review by matt-user — APPROVED** [review]

**Pull request merged**